rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -455,11 +455,25 @@ extern void ppu_register_function_at(u32 addr, u32 size, ppu_function_t ptr)
 	}
 }
 
+extern atomic_t<bool> g_debugger_pause_all_threads_on_bp;
+
 // Breakpoint entry point
 static bool ppu_break(ppu_thread& ppu, ppu_opcode_t)
 {
+	const bool pause_all = g_debugger_pause_all_threads_on_bp;
+
 	// Pause
-	ppu.state.atomic_op([](bs_t<cpu_flag>& state) { if (!(state & cpu_flag::dbg_step)) state += cpu_flag::dbg_pause; });
+	ppu.state.atomic_op([&](bs_t<cpu_flag>& state)
+	{
+		if (pause_all) state += cpu_flag::dbg_global_pause;
```

Comment:
This logic also confuses me, is there any problem with setting both dbg_pause and dbg_global_pause flags? I mean, latter is set by Pause() anyway.

Reply:
Yes, Emu.Resume won't be able to remove dbg_pause.

Reply:
How about automatic resume after resuming __a thread__ waiting on a breakpoint. While normal resume can keep this one thread paused, also simulating old behaviour in the meantime.